XwaDatEditor/XwaDatEditor/MainWindow.xaml.cs

```diff
@@ -449,12 +444,15 @@ private void AddImage_Click(object sender, RoutedEventArgs e)
                 {
                     try
                     {
-                        var image = DatImage.FromFile(group.GroupId, (short)imageId, fileName);
+                        for (int i = 0; i < dialog.FileNames.Length; i++)
+                        {
+                            var image = DatImage.FromFile(group.GroupId, (short)(imageId + i), dialog.FileNames[i]);
 
-                        group.Images.Add(image);
+                            group.Images.Add(image);
 
-                        disp(() => this.DatFile = this.DatFile);
-                        disp(() => this.ImagesList.SelectedIndex = this.ImagesList.Items.Count - 1);
+                            disp(() => this.DatFile = this.DatFile);
+                            disp(() => this.ImagesList.SelectedIndex = this.ImagesList.Items.Count - 1);
```

Comment:
Please move these 2 `disp` lines outside the `for` loop. It is not needed to update the UI after each image is added but only after all new images are added.

Reply:
Done!